src/map_main.cpp

```diff
@@ -345,53 +348,94 @@ int pandora_map(int argc, char *argv[]) {
+    for (auto pan_id_to_node_mapping = pangraph->nodes.begin(); pan_id_to_node_mapping != pangraph->nodes.end(); ++pan_id_to_node_mapping)
+        pangraphNodesAsVector.push_back(pan_id_to_node_mapping->second);
+
+    #pragma omp parallel for num_threads(threads) schedule(dynamic, 10)
+    for (uint32_t i = 0; i < pangraphNodesAsVector.size(); ++i) {
+        //add some progress
+        if (i && i%100==0)
+            BOOST_LOG_TRIVIAL(info) << ((double)i)/pangraphNodesAsVector.size()*100 << "% done";
+
+        //get the node
+        const auto &pangraph_node = pangraphNodesAsVector[i];
 
+        //get the vcf_ref, if applicable
+        std::string vcf_ref;
         if (output_vcf
             and !vcf_refs_file.empty()
             and vcf_refs.find(prgs[pangraph_node->prg_id]->name) != vcf_refs.end()) {
             vcf_ref = vcf_refs[prgs[pangraph_node->prg_id]->name];
         }
 
+        //add consensus path to fastaq
+        std::vector<KmerNodePtr> kmp;
+        std::vector<LocalNodePtr> lmp;
         prgs[pangraph_node->prg_id]->add_consensus_path_to_fastaq(consensus_fq, pangraph_node, kmp, lmp, w, bin, covg);
 
         if (kmp.empty()) {
-            pan_id_to_node_mapping = pangraph->remove_node(pangraph_node);
+            //pan_id_to_node_mapping = pangraph->remove_node(pangraph_node);
+            //mark the node as to remove
+            #pragma omp critical(nodesToRemove)
+            {
+                nodesToRemove.push_back(pangraph_node);
+            }
             continue;
         }
 
         if (output_kg) {
-            pangraph_node->kmer_prg.save(outdir + "/kmer_graphs/" + pangraph_node->get_name() + ".kg.gfa",
+            pangraph_node->kmer_prg_with_coverage.save(outdir + "/kmer_graphs/" + pangraph_node->get_name() + ".kg.gfa",
                                      prgs[pangraph_node->prg_id]);
         }
 
         if (output_vcf) {
+            //TODO: this takes a lot of time and should be optimized, but it is only called in this part, so maybe this should be low prioritized
             prgs[pangraph_node->prg_id]->add_variants_to_vcf(master_vcf, pangraph_node, vcf_ref, kmp, lmp, min_kmer_covg);
         }
 
         if (discover_denovo) {
             const TmpPanNode pangraph_node_components {pangraph_node, prgs[pangraph_node->prg_id], kmp, lmp};
             auto candidate_regions_for_pan_node { find_candidate_regions_for_pan_node(pangraph_node_components, denovo_kmer_size * 2) };
-            candidate_regions.insert(candidate_regions_for_pan_node.begin(), candidate_regions_for_pan_node.end());
+
+            #pragma omp critical(candidate_regions)
+            {
+                candidate_regions.insert(candidate_regions_for_pan_node.begin(), candidate_regions_for_pan_node.end());
+            }
         }
-        ++pan_id_to_node_mapping;
     }
+
+    //remove the nodes marked as to be removed
+    for (const auto &nodeToRemove : nodesToRemove)
```

Comment:
Please make snake_case

Reply:
Done in https://github.com/leoisl/pandora/commit/5b329a78d89eaa0f85592258a8fb378677f71587